lib/rom/memory/dataset.rb

```diff
@@ -46,9 +46,25 @@ def project(*names)
 
       # Sort a dataset
       #
+      # @param [Array<Symbol>] names
+      #   Names of fields to order tuples by
+      #
+      # @option [Boolean] :nils_first (false)
+      #   Whether `nil` values should be placed before others
+      #
       # @api public
       def order(*names)
-        sort_by { |tuple| tuple.values_at(*names) }
+        options = names.last.is_a?(Hash) ? names.pop : {}
+        place   = options[:nils_first] ? -1 : 1
+        compare = ->(a, b) {
+          return  place if a.nil?
+          return -place if b.nil?
+          a <=> b
+        }
+
```

Comment:
Sorry, lemme clarify - a spec that tests a case where there are more than one tuple with more than one attribute with nil :) Makes sense?